internal/eventstore/handler/v2/handler.go

```diff
@@ -0,0 +1,421 @@
+		case <-ctx.Done():
+			t.Stop()
+			return
+		case <-t.C:
+			instances, err := h.queryInstances(ctx, didInitialize)
+			h.log().OnError(err).Debug("unable to query instances")
+
+			var instanceFailed bool
+			scheduledCtx := call.WithTimestamp(ctx)
+			for _, instance := range instances {
+				instanceCtx := authz.WithInstanceID(scheduledCtx, instance)
+				_, err = h.Trigger(instanceCtx)
+				instanceFailed = instanceFailed || err != nil
+				h.log().WithField("instance", instance).OnError(err).Info("scheduled trigger failed")
+
+				for ; err != nil; _, err = h.Trigger(instanceCtx) {
+					instanceFailed = instanceFailed || err != nil
+					h.log().WithField("instance", instance).OnError(err).Info("scheduled trigger failed")
+					if err == nil {
+						break
+					}
+					time.Sleep(h.retryFailedAfter)
+				}
+			}
+
+			if !didInitialize && !instanceFailed {
+				err = h.setSucceededOnce(ctx)
+				h.log().OnError(err).Debug("unable to set succeeded once")
+				didInitialize = err == nil
+			}
+
+			t.Reset(h.requeueEvery)
+		}
+	}
+}
+
+func (h *Handler) subscribe(ctx context.Context) {
+	queue := make(chan eventstore.Event, 100)
+	subscription := eventstore.SubscribeEventTypes(queue, h.eventTypes)
+	for {
+		select {
+		case <-ctx.Done():
+			subscription.Unsubscribe()
+			h.log().Debug("shutdown")
+			return
+		case event := <-queue:
+			events := checkAdditionalEvents(queue, event)
+			solvedInstances := make([]string, 0, len(events))
+			queueCtx := call.WithTimestamp(ctx)
+			for _, e := range events {
+				if instanceSolved(solvedInstances, e.Aggregate().InstanceID) {
+					continue
+				}
+				queueCtx = authz.WithInstanceID(queueCtx, e.Aggregate().InstanceID)
+				_, err := h.Trigger(queueCtx)
+				h.log().OnError(err).Debug("trigger of queued event failed")
+				if err == nil {
+					solvedInstances = append(solvedInstances, e.Aggregate().InstanceID)
+				}
+			}
+		}
+	}
+}
+
+func instanceSolved(solvedInstances []string, instanceID string) bool {
+	for _, solvedInstance := range solvedInstances {
+		if solvedInstance == instanceID {
+			return true
+		}
+	}
+	return false
+}
+
+func checkAdditionalEvents(eventQueue chan eventstore.Event, event eventstore.Event) []eventstore.Event {
+	events := make([]eventstore.Event, 1)
+	events[0] = event
+	for {
+		wait := time.NewTimer(1 * time.Millisecond)
+		select {
+		case event := <-eventQueue:
+			events = append(events, event)
+		case <-wait.C:
+			return events
+		}
+	}
+}
+
+func (h *Handler) queryInstances(ctx context.Context, didInitialize bool) ([]string, error) {
+	query := eventstore.NewSearchQueryBuilder(eventstore.ColumnsInstanceIDs).
+		AllowTimeTravel().
+		AddQuery().
+		ExcludedInstanceID("")
+	if didInitialize {
+		query = query.
+			CreationDateAfter(h.now().Add(-1 * h.handleActiveInstances))
+	}
+	return h.es.InstanceIDs(ctx, h.requeueEvery, !didInitialize, query.Builder())
+}
+
+func (h *Handler) Trigger(ctx context.Context) (_ context.Context, err error) {
```

Comment:
IMO we can already remove the returned context
as we're about to remove the as of system time, it's not necessary anymore
further it might make sense to have tmo methods (trigger / trigger error), where one would directly log the error as currently only handlers are really interested in the error... when querying the errors are anyway only logged, where i'd increase the log level, e.g. warn